dm/dm/master/election_test.go

```diff
@@ -67,6 +68,12 @@ func (t *testElectionSuite) TestFailToStartLeader(c *check.C) {
 	s2 := NewServer(cfg2)
 	c.Assert(s2.Start(ctx), check.IsNil)
 	defer s2.Close()
+	defer cancel() // this cancel must call before s.Close() to avoid deadlock
```

Comment:
why defer cancel three times? L36,L51,L71,L114

Reply:
see this https://github.com/pingcap/tiflow/pull/4556#discussion_r804429264

Reply:
How about defer func(){
    cancel()
    s1.close()
    s2.close()
}()
What you need is to pre declare s1 and s2

Reply:
added in https://github.com/pingcap/tiflow/pull/4566/commits/7c6509b92cd1895c9beec2040e41885ab2812a04